lib/adapter.js

```diff
@@ -820,6 +820,47 @@ module.exports = (function() {
 
     },
 
+    // Count Query Logic
+    count: function(connectionName, table, options, cb) {
+      spawnConnection(connectionName, function __COUNT__(client, cb) {
+
+        // Grab Connection Schema
+        var schema = {};
+        var connectionObject = connections[connectionName];
+        var collection = connectionObject.collections[table];
+
+        Object.keys(connectionObject.collections).forEach(function(coll) {
+          schema[coll] = connectionObject.collections[coll].schema;
+        });
+
+        // Build Query
+        var _schema = connectionObject.schema;
+        var processor = new Processor(_schema);
+
+        // Mixin WL Next connection overrides to sqlOptions
+        var overrides = connectionOverrides[connectionName] || {};
+        var _options = _.cloneDeep(sqlOptions);
+        if(hop(overrides, 'wlNext')) {
+          _options.wlNext = overrides.wlNext;
```

Comment:
[nitpicking]
Is this the only change we do to `_options`? If so I think we can use `_.clone` and improve performance since there won't be any changes to nested values.
[/nitpicking]

Sorry for nitpicking it's just that waterline has too many deep clones and we are trying to reduce them (balderdashy/waterline#989).